Repository/CampaignRepository.php

```diff
@@ -16,10 +16,44 @@
 use DateTime;
 use Doctrine\ORM\EntityRepository;
 
+/**
+ * Class CampaignRepository
+ * @package CampaignChain\CoreBundle\Repository
+ */
 class CampaignRepository extends EntityRepository
 {
 
     /**
+     * get a list of all campaigns
+     *
+     * @return array
+     */
+    public function getCampaigns() {
+
+        return $this->createQueryBuilder('campaign')
+            ->where('campaign.status != :statusBackgroundProcess')
+            ->setParameter('statusBackgroundProcess', Action::STATUS_BACKGROUND_PROCESS)
+            ->orderBy('campaign.startDate', 'DESC')
+            ->getQuery()
+            ->getResult();
+    }
+
+    /**
+     * @param $moduleIdentifier
+     * @return array
+     */
+    public function getCampaignsByModule($moduleIdentifier) {
```

Comment:
@rogamoore The module identifier is not enough to uniquely find a module, it take the bundle name as well.